### PR TITLE
GUVNOR-2452: general improvements on artifact list in order to be eas…

### PR DESCRIPTION
**Pull request by porcelli**
…y to reuse (needed by new exec server ui)

Now bug is fixed, ready to review.

### CODE REVIEW

File: guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/editor/MavenRepositoryPagedJarTable.java [hunk 1]
```diff
@@ -39,84 +41,95 @@
...
+    public void search( String filter ) {
+        presenter().search( filter );
+    }
+
+    public void refresh() {
+        presenter().refresh();
+    }
+
+    private ArtifactListPresenter presenter() {
+        if ( presenter == null ) {
```
Comment (manstis) [inline guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/editor/MavenRepositoryPagedJarTable.java]: There must be a good explanation, but this suggests the View is instantiated without a Presenter; normally we'd work the other way around - Presenter first then View... what's the reasoning here?

Comment (porcelli) [inline guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/editor/MavenRepositoryPagedJarTable.java]: This is not the view's presenter.. this is the ArtifactListPresenter (widget) presenter. And this is related to @AfterInitialization problem that is workarounded here with a lazy init.

Comment (porcelli) [inline guvnor-m2repo-editor/guvnor-m2repo-editor-client/src/main/java/org/guvnor/m2repo/client/editor/MavenRepositoryPagedJarTable.java]: I'd say this implementation is far from ideal... it should be refactored; but as this is supposed to be last 6.x release I'd prefer to make the change as minimium as possible.

**Comment by manstis** [comment]
+1 to merge.

**Comment by csadilek** [comment]
6.4.x: bc63bea598830a02c3ae654c2c577aded6a07e30

**Pull request closed without merging**